processor/PlanExecutor.go

```diff
@@ -18,43 +22,102 @@ func merge(source map[string]interface{}, extra map[string]interface{}) map[stri
 	return source
 }
 
-//ExecutionBranch ...
-type ExecutionBranch interface {
-	Execute(plan core.Plan) error
-}
-
-//PlanExecutor ...
-type PlanExecutor struct {
+//PlanExecutionContext encaosulates the runtime state in order to execute
+//a plan
+type PlanExecutionContext struct {
+	Plan         core.Plan
+	Lists        *ListRingRevolver
 	Config       *config.Configuration
-	Bar          ProgressBar
-	start        time.Time
 	Publisher    telegraph.LinkedPublisher
-	Lists        *ListRingRevolver
-	Plan         core.Plan
 	PlanContext  core.ExtractionResult
 	JobContexts  map[int]core.ExtractionResult
 	StepContexts map[int]map[int]core.ExtractionResult
+	Bar          ProgressBar
 	mutex        *sync.Mutex
+	progress     chan int
+	start        time.Time
 }
 
-//CreatePlanExecutor ...
-func CreatePlanExecutor(config *config.Configuration, bar ProgressBar) *PlanExecutor {
-	return &PlanExecutor{
-		Config:       config,
-		Bar:          bar,
-		Publisher:    telegraph.NewLinkedPublisher(),
-		PlanContext:  core.ExtractionResult{},
-		JobContexts:  map[int]core.ExtractionResult{},
-		StepContexts: map[int]map[int]core.ExtractionResult{},
-		mutex:        &sync.Mutex{},
+func (instance *PlanExecutionContext) execute(cancellation chan struct{}) {
+	instance.workerExecuteJobs(instance.Plan.Jobs, cancellation)
+}
+
+func (instance *PlanExecutionContext) workerExecuteJobs(jobs []core.Job, cancellation chan struct{}) {
```

Comment:
Given that this is only called from the `execute` method, you can ditch the `jobs []core.Job` arg as it is available on the `instance.Plan.Jobs`. This then means that you might as well merge this method into the `execute` method as they would then have the same signature.